tests/unit/test_models/standard_output_tests.py

```diff
@@ -0,0 +1,448 @@
+        elif self.operating_condition == "off":
+            np.testing.assert_array_almost_equal(neg_end_vs_start, 0)
+            np.testing.assert_array_almost_equal(pos_end_vs_start, 0)
+
+    def test_concentration_limits(self):
+        "Test that concentrations do not go below 0 or exceed the maximum."
+
+        np.testing.assert_array_less(-self.c_s_n.entries, 0)
+        np.testing.assert_array_less(-self.c_s_p.entries, 0)
+
+        np.testing.assert_array_less(self.c_s_n.entries, 1)
+        np.testing.assert_array_less(self.c_s_p.entries, 1)
+
+    def test_conservation(self):
+        "Test amount of lithium stored across all particles is constant."
+        # TODO: add an output for total lithium in particles
+
+    def test_concentration_profile(self):
+        """Test that the concentration in the centre of the negative particles is
+        greater than the average concentration in the particle and also that the
+        concentration on the surface of the negative particle is less than the average
+        concentration in the particle. Test opposite is true for the positive
+        particle."""
+        # TODO: add an output for average particle concentration
+
+    def test_fluxes(self):
+        """Test that no flux holds in the centre of the particle. Test that surface
+        flux in the negative particles is less than zero and that the flux on the
+        surface of the positive particles is greater than zeros during a discharge."""
+        # TODO: implement after flux bug fix
+
+    def test_all(self):
+        self.test_concentration_increase_decrease()
+        self.test_concentration_limits()
+        self.test_conservation()
+        self.test_concentration_profile()
+        self.test_fluxes()
+
+
+class ElectrolyteConcentrationTests(BaseOutputTest):
+    def __init__(self, model, disc, solver, operating_condition):
+        super().__init__(model, disc, solver, operating_condition)
+
+        self.c_e = self.get_var("Electrolyte concentration")
+
+        self.c_e_n = self.get_var("Negative electrolyte concentration")
+        self.c_e_s = self.get_var("Separator electrolyte concentration")
+        self.c_e_p = self.get_var("Positive electrolyte concentration")
+
+        # TODO: output average electrolyte concentration
+        # self.c_e_av = self.get_var("Average electrolyte concentration")
+        # self.c_e_n_av = self.get_var("Average negative electrolyte concentration")
+        # self.c_e_s_av = self.get_var("Average separator electrolyte concentration")
+        # self.c_e_p_av = self.get_var("Average positive electrolyte concentration")
+
+        # TODO: fix flux bug
+        # self.N_e = self.get_var("Electrolyte flux")
+
+        # self.N_e_n = self.get_var("Negative electrolyte flux")
+        # self.N_e_s = self.get_var("Separator electrolyte flux")
+        # self.N_e_p = self.get_var("Positive electrolyte flux")
+
+        # self.N_e_hat = self.get_var("Reduced cation flux")
```

Comment:
(just a comment) - not obvious how we get "Negative electrolyte flux" in the DFN, without hacking it (similarly for "Negative electrolyte potential")

Reply:
Ah thank you good to note